python_modules/dagster/dagster/core/host_representation/external_data.py

```diff
@@ -211,29 +211,65 @@ def __new__(cls, error):
 
 
 @whitelist_for_serdes
-class ExternalSensorData(
+class ExternalTargetData(
     namedtuple(
-        "_ExternalSensorData",
-        "name pipeline_name solid_selection mode min_interval description",
+        "_ExternalTargetData",
+        "pipeline_name solid_selection mode",
     )
 ):
     def __new__(
         cls,
-        name,
         pipeline_name,
         solid_selection,
         mode,
+    ):
+        return super(ExternalTargetData, cls).__new__(
+            cls,
+            pipeline_name=check.opt_str_param(pipeline_name, "pipeline_name"),
+            solid_selection=check.opt_nullable_list_param(solid_selection, "solid_selection", str),
+            mode=check.opt_str_param(mode, "mode"),
+        )
+
+
+@whitelist_for_serdes
+class ExternalSensorData(
+    namedtuple(
+        "_ExternalSensorData",
+        "name pipeline_name solid_selection mode min_interval description target_dict",
+    )
+):
+    def __new__(
+        cls,
+        name,
+        pipeline_name=None,
+        solid_selection=None,
+        mode=None,
         min_interval=None,
         description=None,
+        target_dict=None,
     ):
+        if pipeline_name and not target_dict:
+            # handle the legacy case where the ExternalSensorData was constructed from an earlier
+            # version of dagster
+            target_dict = {
+                pipeline_name: ExternalTargetData(
+                    pipeline_name=check.opt_str_param(pipeline_name, "pipeline_name"),
+                    solid_selection=check.opt_nullable_list_param(
+                        solid_selection, "solid_selection", str
+                    ),
+                    mode=check.opt_str_param(mode, "mode"),
+                )
+            }
+
         return super(ExternalSensorData, cls).__new__(
             cls,
             name=check.str_param(name, "name"),
-            pipeline_name=check.opt_str_param(pipeline_name, "pipeline_name"),
-            solid_selection=check.opt_nullable_list_param(solid_selection, "solid_selection", str),
-            mode=check.opt_str_param(mode, "mode"),
+            pipeline_name=None,
+            solid_selection=None,
+            mode=None,
```

Comment:
the safest thing to do would be to write out the values if single target so old versions reading can still process